Formula/postgresql@12.rb

```diff
@@ -81,22 +81,100 @@ def post_install
     return if ENV["CI"]
 
     (var/"log").mkpath
-    (var/"postgres").mkpath
-    unless File.exist? "#{var}/postgres/PG_VERSION"
-      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", "#{var}/postgres"
+    versioned_postgresql_data_dir.mkpath
+    unless versioned_data_dir_exists?
+      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", versioned_postgresql_data_dir
     end
   end
 
+  # Previous versions of this formula used the same data dir as the regular
+  # postgresql formula. So we check whether the versioned data dir exists
+  # and has a PG_VERSION file, which should indicate that the versioned
+  # data dir is in use. Otherwise, returns the old data dir path.
+  def postgresql_datadir
+    if versioned_data_dir_exists?
+      versioned_postgresql_data_dir
+    else
+      old_postgresql_data_dir
+    end
+  end
+
+  def versioned_postgresql_data_dir
+    "#{var}/#{name}"
+  end
+
+  def old_postgresql_data_dir
+    "#{var}/postgres"
+  end
+
+  # Same as with the data dir - use old log file if the old data dir
+  # is version 12
+  def postgresql_log_path
+    if versioned_data_dir_exists?
+      "#{var}/log/#{name}"
+    else
+      "#{var}/log/postgres"
+    end
+  end
+
+  def versioned_data_dir_exists?
+    File.exist?("#{versioned_postgresql_data_dir}/PG_VERSION")
+  end
+
+  def conflicts_with_postgresql_formula?
+    Formula["postgresql"].any_version_installed?
+  end
+
+  # Figure out what version of PostgreSQL the old data dir is
+  # using
+  def old_postgresql_datadir_version_12?
+    File.exist?("#{old_postgresql_data_dir}/PG_VERSION") &&
+      File.read("#{old_postgresql_data_dir}/PG_VERSION").chomp == "12"
```

Comment:
```suggestion
    pg_version = old_postgresql_data_dir/"PG_VERSION"
    pg_version.exist? && pg_version.read.chomp == "12"
```

Might be nicer still to extract the `@12` from the name and use that.

Reply:
Addressed in 29942ac6d6.

I approached it a bit differently: This method now returns the version string from the `PG_VERSION` file (or nil if it doesn't exist), and then in the `caveats` method, I extract the formula's version number and compare the two.